### ISSUE TITLE
Permit [SameObject] for nullable types?

### ISSUE THREAD

**Issue by jcjones** [heycam/webidl#559]
WebAuthn made a previously-[SameObject] attribute userHandle nullable (in https://github.com/w3c/webauthn/commit/3b2a1d141cbd8f2954f073a6b6598d954398a986). @smaug---- points out that ["SameObject applies to interface and objects only, not nullable types"](https://phabricator.services.mozilla.com/D1337#inline-5985) but since this sort of makes sense, perhaps WebIDL should change to permit the behavior.

**Comment by annevk** [comment]
It either always returns null or always returns an object? In any event, given that we want to replace [SameObject] in #212 I'm not sure we should extend it now.

**Comment by jcjones** [comment]
Yeah, that's right. It's always either null, or a cached object, basically.
